Lesson2/src/main/java/pl/solvd/university/administration/Deanery.java

```diff
@@ -0,0 +1,80 @@
+package pl.solvd.university.administration;
+
+import pl.solvd.university.people.Applicant;
+import pl.solvd.university.people.Student;
+import pl.solvd.university.state.ReturnDocuments;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static pl.solvd.university.Main.*;
+
+public class Deanery implements Comparator<Applicant> {
+
+    @Override
+    public int compare(Applicant o1, Applicant o2) {
+        if (o2.getSum() == o1.getSum()) {
+            return o2.getFirstLastName().compareTo(o1.getFirstLastName());
+        } else
+            return o2.getSum() - o1.getSum();
+    }
+
+    public static void sortByGrades(List<Applicant> sortedList, int NumOfBudgetPlaces, int NumOfPaidPlaces, String getSpecialty) throws Exception {
```

Comment:
it doesn't work if I split the methods into sortByGrades() and findApplicantsForAdmission() . It is not possible to transfer a sorted list from sortByGrade() to another method: the list sortedApplicantsCopy asks to be initialized as null from beginning and then does not sort as needed. At least for the remaining time, I will not solve this issue. Therefore, I have to transfer all the necessary data at once, and if it helps, I will change the name

Reply:
Changing the name is also an option. The thing is that on real project you will not work alone, and the goal is to let your colleagues to save as much time as possible by writing self-descriptive code. In case name doesn't fit required parameters, other worker will have to look into function to see what is going on here and verify it does exactly what he wants this function to do, if not - he will implement his own function. But if the name and parameters are matched, but function does smth not exepected - this might kill someones logic of appilcation and they will spend some time debugging to understand what is happening, this is not expected at all.

Reply:
Got it. This comment explains a lot

Reply:
Done